vendor: bump Pebble to 86efa0d0, and bump related deps

sumeerbhola opened the pull request:
86efa0d0 Merge pull request #1150 from petermattis/pmattis/compact-metrics
35a4e5d0 db: add per-compaction type counts to Metrics.Compact
39fe91db Merge pull request #1147 from petermattis/pmattis/compact-events
7a7a1118 db: flesh out CompactionInfo.Reason
e1d9dc9c Merge pull request #1148 from petermattis/pmattis/options
698f80b8 db: add missing options to the OPTIONS file
3ebefb17 Merge pull request #1135 from petermattis/pmattis/flush-empty-key
6933a5a7 sstable: fix sstable.Writer key order checks for nil keys
10e81784 db: add IteratorStats, for counting the work done in Iterator

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65529)
<!-- Reviewable:end -->

itsbilal reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 5 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65529#-:-MaATn8Q1etYWjAtq3qw:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @jbowens)



<!-- Sent from Reviewable.io -->

sumeerbhola committed:
vendor: bump Pebble to 86efa0d0, and bump related deps

86efa0d0 Merge pull request #1150 from petermattis/pmattis/compact-metrics
35a4e5d0 db: add per-compaction type counts to Metrics.Compact
39fe91db Merge pull request #1147 from petermattis/pmattis/compact-events
7a7a1118 db: flesh out CompactionInfo.Reason
e1d9dc9c Merge pull request #1148 from petermattis/pmattis/options
698f80b8 db: add missing options to the OPTIONS file
3ebefb17 Merge pull request #1135 from petermattis/pmattis/flush-empty-key
6933a5a7 sstable: fix sstable.Writer key order checks for nil keys
10e81784 db: add IteratorStats, for counting the work done in Iterator

Release note: None

sumeerbhola commented:
TFTR!

I pinned `honnef.co/go/tools` as discussed here https://cockroachlabs.slack.com/archives/CJ0H8Q97C/p1621630442008900?thread_ts=1605298982.230500&cid=CJ0H8Q97C

sumeerbhola commented:
bors r+
